azkaban-web-server/src/main/java/azkaban/webapp/WebMetrics.java

```diff
@@ -38,8 +37,13 @@
   // How long does user log fetch take when user call fetch-log api.
   private final AtomicLong logFetchLatency = new AtomicLong(0L);
 
-  WebMetrics() {
-    this.registry = MetricsManager.INSTANCE.getRegistry();
+  @Inject
+  WebMetrics(final MetricRegistry registry) {
+    // Add unit test coverage to guarantee no duplication
+    // Seeing this error after guicifying, but before binding as singleton in AzkabanWebServerModule:
```

Comment:
IMHO, I prefer annotating the class itself with singleton if the class is designed to be used as singleton everywhere. It's less code and the information is centralized. Otherwise, readers have to check the module files too to find this information.